cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -319,30 +317,29 @@ private void expireOldSessions() {
    * the User-Task-ID from the request header and check if there is any UserTask with the same User-Task-ID.
    * If no User-Task-ID is passed then the {@link HttpSession} is used to fetch the User-Task-ID.
    *
-   * @param httpServletRequest the HttpServletRequest to fetch the User-Task-ID and HTTPSession.
+   * @param handler the HttpServletRequest to fetch the User-Task-ID and HTTPSession.
    * @return UUID of the user tasks or null if user task doesn't exist.
    */
-  public UUID getUserTaskId(HttpServletRequest httpServletRequest) {
-    String userTaskIdString = httpServletRequest.getHeader(USER_TASK_HEADER_NAME);
-
+  public UUID getUserTaskId(HttpFrameworkHandler handler) {
+    String userTaskIdString = handler.getHeader(USER_TASK_HEADER_NAME);
     UUID userTaskId;
     if (userTaskIdString != null && !userTaskIdString.isEmpty()) {
       // valid user task id
       userTaskId = UUID.fromString(userTaskIdString);
     } else {
-      SessionKey sessionKey = new SessionKey(httpServletRequest);
+      SessionKey sessionKey = new SessionKey(handler);
       synchronized (_sessionKeyToUserTaskIdMap) {
         userTaskId = _sessionKeyToUserTaskIdMap.get(sessionKey);
       }
     }
-
     return userTaskId;
   }
 
    synchronized void checkActiveUserTasks() {
     Iterator<Map.Entry<UUID, UserTaskInfo>> iter = _uuidToActiveUserTaskInfoMap.entrySet().iterator();
     while (iter.hasNext()) {
       Map.Entry<UUID, UserTaskInfo> entry = iter.next();
+      System.out.println("asdasdasd");
```

Comment:
Well it's [more secure than log4j2](https://blog.cloudflare.com/inside-the-log4j2-vulnerability-cve-2021-44228/) that's for sure but let's remove this :)